### ISSUE TITLE
Octal Permissions Rule, false positive ?

### ISSUE THREAD

**Issue by Tiduster** [willthames/ansible-lint#235]
Hello !

# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.2.0.0
ansible-lint 3.4.9
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour
- name: Ensure destination directories exist and have proper permissions

```yml
  file:
    name: /etc/ssl/private
    state: directory
    owner: root
    group: ssl-cert
    mode: 0710
```

No warning should be triggered

# Actual Behaviour

This task trigger the rule "ANSIBLE0009" because "file permission modes don't have write or execute bit set when read bit is not set", which isn't true for a folder.

On another subject I don't understand why string and int are treated differently:

```python
if isinstance(mode, six.string_types) and self.mode_regex.match(mode):
if isinstance(mode, int):
```
IMHO, we should convert string mode into int and use the same check for both.

I can make a PR if you agree with this bug report.

Regards,
